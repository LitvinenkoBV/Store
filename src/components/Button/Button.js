import React from "react";
import "./Button.scss";
import PropTypes from "prop-types";

const Button = (props) => {
  const { text, style, openModal, dataModalId, product } = props;
  return (
    <button
      data-modal-id={dataModalId}
      className="button__open-modal"
      style={style}
      onClick={() => {
        openModal(dataModalId, product);
      }}
    >
      {text}
    </button>
  );
};

Button.propTypes = {
  dataModalId: PropTypes.number,
  openModal: PropTypes.func,
  product: PropTypes.object,
  style: PropTypes.object,
  text: PropTypes.string,
};
Button.defaultProps = {
  text: "Add to card?",
};

export default Button;
