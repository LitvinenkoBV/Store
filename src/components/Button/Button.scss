.button {
  display: flex;
  justify-content: space-around;
  &__open-modal {
    padding: 10px;
    font-size: 16px;
    color: white;
    border-radius: 5px;
  }
}
