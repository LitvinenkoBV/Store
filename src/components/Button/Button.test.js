import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Button from "./Button";

describe("<Button/> component", () => {
  const { container } = render(<Button></Button>);
  it("should render", () => {
    expect(container.firstChild).toMatchSnapshot();
  });
  it("should render button tag", () => {
    const { container } = render(<Button></Button>);
    screen.getByRole("button");
  });

  it("should render button with text", () => {
    const { container } = render(<Button text="Add to Card"></Button>);
    screen.getByText(/Add to Card/);
  });

  it("should pass onClick if button is clicked", () => {
    const handleClick = jest.fn();
    render(<Button openModal={handleClick}></Button>);
    userEvent.pointer({
      keys: "[MouseLeft]",
      target: screen.getByRole("button"),
    });
    expect(handleClick).toHaveBeenCalled();
  });
});
