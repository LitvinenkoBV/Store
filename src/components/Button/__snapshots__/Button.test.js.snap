// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> component should render 1`] = `
<button
  class="button__open-modal"
>
  Add to card?
</button>
`;
